### DISCUSSION TITLE
0.10.11-rc: Data API builder for Azure Databases

### DISCUSSION THREAD

**Discussion by seantleonard** [Azure/data-api-builder#1917] [Announcements]
#### DAB ENGINE

* 40dab94d85b68f538e336e0b732f8cf73fca026e In-Memory Cache  (#1881) [ #1618, #1617, #1801, #1865 ]
* 0829d859301e9611c501b562cfc85e60e20b485a Handling datetime types (#1636)
* ed4fc13b7c9805729784a54f8993f9bc8a40ed03 DataWarehouse query support for data-api-builder (#1873)
* 8676ed46d619ed65a85ff18f1d744fae815d0e33 Adding support for query parameter in Swagger UI (#1769) [ #1572 ]
* 4080518357d6f81efaf0d9771a82807a8ccd28df Hot Reload of Runtime Section (#1726)
* 0cf5d64b0e40ea733676f14017049e4d4b71aa1b Adding support for providing Auth headers via Swagger UI (#1788)
* 0092cad2d337078509561a1b96f26012b644185a Addition of new database type DWSQL. (#1864)
* 3176820ac95fe973ad51bca5cdc1599c7b4dc088 Data Api Builder support for multiple databases. (#1709)

#### DAB CLI

* e26d50717753272ca797c45c19e7aad6b6e52f91 Add `dab validate` CLI command (#1856) [ #1845 ]

#### DAB Runtime Config

* 36ab82ad068d7611d9a967cc20be25d0fb6f5e1c Add Cache properties to runtime config file with JSON Serialize/Deserialize (#1865) [ #1618, #1617 ]

### Other enhancements and bug fixes

* fffcd703ef85b9df287305e1e4a8bf8159f5ebe2 Removed hash from application name (fix #1759) (#1863)
* ea2ab9b82781993f2b3d1f91465a5f0d9149854b Adds helpful error message for mutations executed without read permissions - SQL database types (#1874) [ #1464 ]
* bcb891b60e5631e6afa570424c08c0192feb45d2 Fix base64 encoding of $after query parameter value in NextLink (#1895)
* d1fa539e2440159e11a27be1f84c6a7180288b47 Change the form of our NextLink used for pagination cursors (#1851)
* 49796db8d85210d75730b6709d88aae3df27f8cc Update DAB runtime config JSON schema (#1845)
* 7efa7f4823382e346205ab5e3617080ec9fb0bea Update dab.draft.schema.json (#1782)
* c09e08a01fd9f3d828668813b4f3adff6709cfac Copying stored proc parameters in the config no longer required in some circumstances. (#1847)
* 7460003d78377fc1707a997080c5f201c2dac729 Attempt to create OpenAPI document only if rest is enabled (#1872)
* 8707c631c2a31a0c801a336f69e7b020d70f4c05 Add telemetry client only if it is enabled (#1868)
* d885e599e1371a9c36975678156b3dadc4026cc1 Enhance shutdown logs with extra details (#1821) [ #1763 ]
* 8ab7f369eb64f9aa8fe2acbee27952bf3ffd8cf3 Access Token needs to be set according to the datasource name. (#1846)
* 00842f6ca4769799bbeba2abfbde0973be7a0a38 [Configuration] Make optional object sub properties nullable (#1823) [ #1402, #1684, #1737 ]
* a12f39fc8f36464b001acc3de74a957567cfa5c3 Handling missing environment variables cleaner for SWA (#1825) [ #1820 ]
* a60b5982844970d45ed7987e68abaddad5fb13ea Update dependencies (#1828)
* 020792ee178209416e51c804c245e4af7e2b2304 Upgrade HotChocolate (#1854)
* 49ac53ac05910128358b620b4498c141d03d7ebc Introduces a change in the REST POST, PATCH and PUT API response construction (#1731)
* ca5fa1463a38b3a0b5aebf31a4bff8aa64cebe73 Move to file scoped namespace project Core part 1 (#1744)
* 9ad8b34399513e45337ee4babca176e59ef8af23 Ensuring correct population of second constructor in runtimeconfig. (#1772)
* bd6c94314b16cacfde974d3065cc5791560669ae Unhandled Exception for malformed connection string bug fix (#1741)
* 51adeb290138e39fec7d985a604459fff6781bc0 Fix setting CosmosDB configuration at runtime by checking whether the graphql schema is set (#1743) [ #1739 ]
* 038c17d57c6d0ac4406b090e5ca84a56732a573c Docker file now used Azure Mariner Linux (#1786) [ #1762 ]

<details><summary><b>See More</b></summary>

### Documentation updates

* da8b47b83c503f9f494c9fd853e15bd2a449cdf1 REST and GraphQL Query Caching Design Doc (#1801)
* 92dde3ff0feacef49db5a68bc8bab6966be2e3a7 ACA sample improvements (#1858)
* 2471e18489127ca94c11c665c016ab6e2223af90 Update README.md (#1882)
* ac2f79e3e97cc5aab03affcb6154fe4d30e3d9f1 Add design doc and diagram for Hot Reload feature (#1811)
* a0b64c0196c9f8ec722bafdb0d1e4c9bc8a1a016 Update CONTRIBUTING.md (#1776)
* 8b6564a5c69c98c9a506fb3730e14ba69eb4f635 Update README.md (#1758)

### Test infrastructure updates

* ee1f78c65ad0d92b937beb509ff5514aed91e9cd Eliminate generation of config files multiple times in the integration test pipelines (#1910) [ #1909 ]
* 6c213a57f48177be2c01c837175243ed4b208b87 Adding pipeline for DW testing. (#1877)
* fdcf2cc7259ec9f429fe4d7e6226b6a79da950cf Fixing dw-sql pipeline and enabling DW Query Builder to handle Unicode characters (#1907) [ #1901 ]
* 74cb4b4dc427efd2c20d08dbb61d0ca1dc61f106 Configured CosmosDB Local Emulator for running integration tests on the Windows agent (#1876)
* bec778fa92c0902f2789594300cbe66c00cd02a8 Update minor version in build ci pipeline config to 0.10 in MAIN (#1742)

This list of changes was [auto generated](https://msdata.visualstudio.com/CosmosDB/_build/results?buildId=112889372&view=logs). The change list was manually rearranged into distinct sections
to categorize updates.</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/Azure/data-api-builder/releases/tag/v0.10.11-rc'>0.10.11-rc: Data API builder for Azure Databases</a>.</em>

**Comment by yorek** [comment]
For anyone willing to try the RC, it can be installed via 

```
dotnet tool update Microsoft.DataApiBuilder -g --prerelease
```

at any time you need to go back to the stable version, just uninstall the pre-release version

```
dotnet tool uninstall Microsoft.DataApiBuilder -g 
```

and install the current release

```
dotnet tool install Microsoft.DataApiBuilder -g 
```
